test/async_copy.jl

```diff
@@ -0,0 +1,25 @@
+using KernelAbstractions, Test, CUDAapi
+if has_cuda_gpu()
+    using CuArrays, CUDAdrv
+    CuArrays.allowscalar(false)
+end
+
+function copy_test(backend, ArrayT, M)
+    A = ArrayT(rand(Float64, M))
+    B = ArrayT(rand(Float64, M))
+
+    a = Array{Float64}(undef, M)
+    event = async_copy!(backend, a, B)
+    event = async_copy!(backend, A, a, dependencies=event)
+    wait(event)
+
+    @test isapprox(a, Array(A))
+    @test isapprox(a, Array(B))
+end
+
+M = 1024
+
+if has_cuda_gpu()
```

Comment:
Maybe add testsets around these.

Also, I feel there should be a test for all possible copies: CPU-> CPU, GPU->CPU, GPU->GPU (with pin), CPU->GPU.

Reply:
Interestingly the tests to hit all those code-paths already!